ibis/impala/compiler.py

```diff
@@ -368,36 +368,38 @@ def unary(func_name):
     return fixed_arity(func_name, 1)
 
 
-def _reduction_format(translator, func_name, arg, where):
+def _reduction_format(translator, func_name, arg, args, where):
     if where is not None:
-        case = where.ifelse(arg, ibis.NA)
-        arg = translator.translate(case)
-    else:
-        arg = translator.translate(arg)
+        arg = where.ifelse(arg, ibis.NA)
 
-    return '{}({})'.format(func_name, arg)
+    return '{}({})'.format(
+        func_name, ', '.join(map(translator.translate, [arg] + list(args)))
+    )
 
 
 def _reduction(func_name):
     def formatter(translator, expr):
         op = expr.op()
 
         # HACK: support trailing arguments
-        arg, where = op.args[:2]
+        where = op.where
+        args = [arg for arg in op.args if arg is not where]
 
-        return _reduction_format(translator, func_name, arg, where)
+        return _reduction_format(
+            translator, func_name, args[0], args[1:], where
+        )
     return formatter
 
 
 def _variance_like(func_name):
     func_names = {
-        'sample': func_name,
+        'sample': '{}_samp'.format(func_name),
```

Comment:
I changed these names so that we could reuse the same translation mechanism in the BigQuery backend.